applications/mne_anonymize/fiffanonymizer.cpp

```diff
@@ -1005,118 +679,133 @@ void FiffAnonymizer::setBruteMode(bool bFlag)
 
-void FiffAnonymizer::setMeasurementDayOffset(int iMeasDayOffset)
+void FiffAnonymizer::setMeasurementDay(const QDate& d)
 {
-    m_bUseMeasurementDayOffset = true;
-    m_iMeasurementDayOffset = iMeasDayOffset;
+    m_dMeasurementDate = QDateTime(d,QTime(1,1,0));
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::setSubjectBirthday(const QString& sSubjBirthday)
+void FiffAnonymizer::setMeasurementDay(const QString& sMeasDay)
 {
-    m_dateSubjectBirthday = QDateTime(QDate::fromString(sSubjBirthday),QTime(1, 1, 0));
+
+    m_dMeasurementDate = QDateTime(QDate::fromString(sMeasDay,"ddMMyyyy"),QTime(1,1,0));
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::setSubjectBirthdayOffset(int iSubjBirthdayOffset)
+QDateTime FiffAnonymizer::getMeasurementDate() const
 {
-    m_bUseSubjectBirthdayOffset = true;
-    m_iSubjectBirthdayOffset = iSubjBirthdayOffset;
+    return m_dMeasurementDate;
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::setDeleteInputFileAfter(bool bFlag)
+void FiffAnonymizer::setMeasurementDayOffset(const int iMeasDayOffset)
 {
-    m_bDeleteInputFileAfter = bFlag;
+    m_bUseMeasurementDayOffset = true;
+    m_iMeasurementDayOffset = iMeasDayOffset;
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::setDeleteInputFileAfterConfirmation(bool bFlag)
+void FiffAnonymizer::setSubjectBirthday(const QString& sSubjBirthday)
 {
-    m_bDeleteInputFileConfirmation = bFlag;
+    m_dSubjectBirthday = QDateTime(QDate::fromString(sSubjBirthday,"ddMMyyyy"),QTime(1, 1, 0));
 }
 
 //=============================================================================================================
 
-QString FiffAnonymizer::generateRandomFileName()
+void FiffAnonymizer::setSubjectBirthday(const QDateTime& sSubjBirthday)
 {
-    QString randomFileName("mne_anonymize_");
-    const QString charPool("abcdefghijklmnopqrstuvwxyz1234567890");
-    const int randomLength(8);
+    m_dSubjectBirthday = QDateTime(sSubjBirthday);
+}
 
-    for(int i=0;i<randomLength;++i) {
-        int p=qrand() % charPool.length();
-        randomFileName.append(charPool.at(p));
-    }
+//=============================================================================================================
 
-    return randomFileName.append(".fif");
+QDateTime  FiffAnonymizer::getSubjectBirthday()
+{
+    return m_dSubjectBirthday;
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::deleteInputFile()
+void FiffAnonymizer::setSubjectBirthdayOffset(const int iSubjBirthdayOffset)
```

Comment:
`const` is not needed here since the passed value is not a reference

Reply:
Rethinking it again, and reading about it. I've come to the conclusion that it is better to not use the const modifier. It sort-of-breaks my dear encapsulation. I like that. Thanks Lorenz.